python/pacemaker/_cts/clustermanager.py

```diff
@@ -24,6 +24,23 @@
 from pacemaker._cts.remote import RemoteFactory
 from pacemaker._cts.watcher import LogWatcher
 
+# Throughout this file, pylint has trouble understanding that EnvFactory
+# and RemoteFactory are singleton instances that can be treated as callable
+# and subscriptable objects.  Various warnings are disabled because of this.
+# See also a comment about self._rsh in environment.py.
+# pylint: disable=unsubscriptable-object
+
+# pylint doesn't understand that self._rsh is callable (it stores the
+# singleton instance of RemoteExec, as returned by the getInstance method
+# of RemoteFactory).  It's possible we could fix this with type annotations,
+# but those were introduced with python 3.5 and we only support python 3.4.
+# I think we could also fix this by getting rid of the getInstance methods,
+# but that's a project for another day.  For now, just disable the warning.
+# pylint: disable=not-callable
```

Comment:
yeah eventually i'd love to just get rid of the singletons altogether and pass arguments around instead

Reply:
Yep, this was very much on my todo list for after everything's in the _cts module.